src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/IFileSystem.cs

```diff
@@ -12,6 +12,18 @@ namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Linux;
 /// </summary>
 internal interface IFileSystem
 {
+    /// <summary>
+    /// Checks for file existence.
+    /// </summary>
+    /// <returns> True/False.</returns>
+    bool Exists(FileInfo fileInfo);
+
+    /// <summary>
+    /// Get directory names on the filesystem based on the provided pattern.
+    /// </summary>
+    /// <returns>string.</returns>
+    string[] GetDirectoryNames(string directory, string pattern);
```

Comment:
Even though this is an internal class I'd caution against using mutable arrays as return values. This could lead to some gnarly hard to track bugs.
Could this be changed to something like `IList<string>` or `IEnumerable<string>`?

Reply:
Makes sense, will do the changes in another PR